spring-cloud-function-web-handler/src/main/java/io/sk8s/handler/spring/web/invoker/FunctionConfiguration.java

```diff
@@ -49,56 +60,71 @@
 	private ResourceLoader contextResourceLoader;
 
 	@Autowired
+	private DelegatingResourceLoader delegatingResourceLoader;
+
+	@Autowired
 	private FunctionProperties properties;
 
 	@Bean
-	@SuppressWarnings({ "unchecked", "rawtypes" })
-	public Function<Flux<String>, Flux<String>> function() {
-		URLClassLoader classLoader = null;
-		List<URL> urls = new ArrayList<>();
-		try {
-			String[] locations = StringUtils.tokenizeToStringArray(this.properties.getUri(), ",");
-			for (String location : locations) {
-				if (location.startsWith("maven:")) {
-					MavenResourceLoader mavenResourceLoader = new MavenResourceLoader(new MavenProperties());
-					MavenResource resource = (MavenResource) mavenResourceLoader.getResource(location);
-					urls.add(new URL("file:" + resource.getFile().getAbsolutePath()));
-				}
-				else {
-					urls.add(this.contextResourceLoader.getResource(location).getURL());
-				}
-			}
-			// TODO: add a 'dependencies' property for additional JARs
-			classLoader = new URLClassLoader(urls.toArray(new URL[urls.size()]), this.getClass().getClassLoader());
-			String[] classnames = StringUtils.tokenizeToStringArray(this.properties.getClassname(), ",");
-			Assert.isTrue(classnames.length == locations.length, "locations and classnames must have the same number of values");
-			Function function = null;
-			for (int i = 0; i < classnames.length; i++) {
-				Class<?> clazz = classLoader.loadClass(classnames[i]);
-				Function currentFunction = (Function) clazz.newInstance();
-				if (i == 0) {
-					function = currentFunction;
-				}
-				else {
-					function = function.andThen(currentFunction);
-				}
-			}
-			return (!FunctionFactoryUtils.isFluxFunction(function))
-					? new FluxFunction<String, String>((Function<String, String>) function)
-					: (Function<Flux<String>, Flux<String>>) function;
+	@ConfigurationProperties("maven")
+	public MavenProperties mavenProperties() {
+		return new MavenProperties();
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(DelegatingResourceLoader.class)
+	public DelegatingResourceLoader delegatingResourceLoader(MavenProperties mavenProperties) {
+		Map<String, ResourceLoader> loaders = new HashMap<>();
+		loaders.put(MavenResource.URI_SCHEME, new MavenResourceLoader(mavenProperties));
+		return new DelegatingResourceLoader(loaders);
+	}
+
+	@Bean
+	public Object function() throws Exception {
+		URL[] urls = Arrays.stream(properties.getUri())
+				.map(toResourceURL())
+				.toArray(URL[]::new);
+
+		try (URLClassLoader cl = new URLClassLoader(urls, /* explicit null parent */null)) {
+			return Arrays.stream(properties.getClassName())
+					.map(name -> (Object) BeanUtils.instantiateClass(ClassUtils.resolveClassName(name, cl)))
+					.sequential()
+					.reduce(this::compose).get();
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private Object compose(Object a, Object b) {
```

Comment:
Also, did not mention this otp, but I guess something similar to this could move to s-c-function